### PR TITLE
Center window on startup

### PR DESCRIPTION
**Pull request by jeremycostanzo**
Resolves: #1030 
## What kind of change does this PR introduce?
- Feature

## Did this PR introduce a breaking change? 
_A breaking change includes anything that breaks backwards compatibility either at compile or run time._
- Yes, please list breaking changes
- I changed the default geometry from 50 to 30, I am not sure whether it could break things

### CODE REVIEW

**Comment by jeremycostanzo** [comment]
I changed the default geometry to make the window larger than it is high.
I did it because with default settings, the window was too high for my 1920/1080 screen on windows, but I can remove the change from this PR so that can discuss it in another issue ^^.

File: src/window/window_wrapper/mod.rs [hunk 1]
```diff
@@ -339,10 +341,32 @@ pub fn create_window(
...
 
 fn is_already_resized(size: PhysicalSize<u32>) -> bool {
     size != PhysicalSize::from((800, 600))
 }
+
+fn center_window(handle: winit::monitor::MonitorHandle, window: &Window) {
+    let outer_size = window.outer_size();
+    let monitor_size = handle.size();
+    let window_x = (monitor_size.width as i32 - outer_size.width as i32) / 2;
+    let window_y = (monitor_size.height as i32 - outer_size.height as i32) / 2;
```
Comment (LoipesMas) [inline src/window/window_wrapper/mod.rs:367]: Can this be `(monitor_size.width - outer_size.width) as i32 / 2`?
Should look a bit cleaner.

Comment (jeremycostanzo) [reply src/window/window_wrapper/mod.rs:367]: I added the `as i32` afterwards, because both `monitor_size.width` and `outer_size.width` are usize, so the substraction would fail if `outer_size.width` is larger than `monitor_size.width`.

With the `as i32`, it does not fail and the window is centered.

Although I agree that it is not satisfying to have it 4 times in 2 lines, so don't hesitate if you can think of a better implementation :).

Comment (LoipesMas) [reply src/window/window_wrapper/mod.rs:367]: Ah, they're usize. That changes things.
I'm not sure what the expected behaviour should be when `outer_size` is larger than `monitor_size`.
Maybe [saturating_sub](https://doc.rust-lang.org/stable/std/primitive.usize.html#method.saturating_sub) could be used? So then if `outer_size` is larger than `monitor_size` it will "snap" to top-left corner (I think).

Comment (LoipesMas) [reply src/window/window_wrapper/mod.rs:367]: Maximizing it seems reasonable

Comment (jeremycostanzo) [reply src/window/window_wrapper/mod.rs]: It should be better now

**Comment by LoipesMas** [comment]
Looks good. Can't test it myself (tiling window manager), so I'll trust you (or if someone else tests this) that it works.

I think we should leave the default geometry out of this PR, since this wasn't related to the issue.

Also, now that I think of it, should it always center the window? Don't some window managers open windows in particular positions (i.e. kinda like tiling, or opening where it was closed (or is it always handled on application side))?
And always centering could mess with it.

**Comment by Kethku** [comment]
Also how does this interact with https://github.com/neovide/neovide/issues/887 which tries to remember the position of the window on startup

**Comment by last-partizan** [comment]
I don't think neovide should center it's window by default (or at all).

Right now window manager takes care of window positions, and doing this job just fine. (In my case - gnome places windows in less cluttered part of the screen, and it's great behaviour).

> With the latest stable version, the window was centered at startup. This leads me to believe that this is a regression.

@Kethku does this looks like regression? i don't remember it working that way.

I think, it's better not to touch initial position and instead implement remembering last position. If it was at center - it will be restored.

But, if this really adds value for some people, it should be added behind a flag, so it would not interfere with default behaviour.

File: src/settings/window_geometry.rs [hunk 2]
```diff
@@ -24,7 +24,7 @@ pub fn try_to_load_last_window_size() -> Result<Dimensions, String> {
 }
 pub const DEFAULT_WINDOW_GEOMETRY: Dimensions = Dimensions {
     width: 100,
-    height: 50,
+    height: 30,
```
Comment (last-partizan) [inline src/settings/window_geometry.rs:27]: I think this is unrelated, and better be reverted.

You always can create alias for your default size.

**Comment by Kethku** [comment]
@last-partizan I believe this was an unintended side effect of using sdl2. Now that things are on winit, we have more control.

Also a bunch of changes just went in to the way we persist window position and size, so I'm inclined to say we should abandon this PR or lock the changes behind a setting (and or environment variable).

@jeremycostanzo what do you think?

**Comment by jeremycostanzo** [comment]
@Kethku Yes I understand, let's close this then!
